CMakeLists.txt

```diff
@@ -1,89 +1,55 @@
-SET (OPENNS_SRC_DIR                             ${OPENNS_SOURCE_DIR}/api)
-SET (OPENNS_EXAMPLES_SOURCE_DIR                 ${OPENNS_SOURCE_DIR}/examples)
-if(WIN32)
-	SET (CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE     ${OPENNS_SOURCE_DIR}/bin)
-	SET (CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG       ${OPENNS_SOURCE_DIR}/bin)
-	SET (CMAKE_ARCHIVE_OUTPUT_DIRECTORY_RELEASE     ${OPENNS_SOURCE_DIR}/lib)
-	SET (CMAKE_ARCHIVE_OUTPUT_DIRECTORY_DEBUG       ${OPENNS_SOURCE_DIR}/lib)
+# SWIG: use standard target name.
+if(POLICY CMP0078)
+    cmake_policy(SET CMP0078 NEW)
 endif()
 
-#-----------------------------------------------------------------------------
-# parse the full version number from bag.h and include in OPENNS_VERS_INFO
-#-----------------------------------------------------------------------------
-FILE (READ ${OPENNS_SRC_DIR}/bag.h _bag_h_contents)
-STRING (REGEX REPLACE ".*#define[ \t]+BAG_VER_MAJOR[ \t]+([0-9]*).*$"
-    "\\1" OPENNS_VERS_MAJOR ${_bag_h_contents})
-STRING (REGEX REPLACE ".*#define[ \t]+BAG_VER_MINOR[ \t]+([0-9]*).*$"
-    "\\1" OPENNS_VERS_MINOR ${_bag_h_contents})
-STRING (REGEX REPLACE ".*#define[ \t]+BAG_VER_REVISION[ \t]+([0-9]*).*$"
-    "\\1" OPENNS_VERS_RELEASE ${_bag_h_contents})
-
-#-----------------------------------------------------------------------------
-SET (OPENNS_PACKAGE "openns")
-SET (OPENNS_PACKAGE_NAME "OPENNS")
-SET (OPENNS_PACKAGE_VERSION "${OPENNS_VERS_MAJOR}.${OPENNS_VERS_MINOR}")
-SET (OPENNS_PACKAGE_VERSION_MAJOR "${OPENNS_VERS_MAJOR}.${OPENNS_VERS_MINOR}")
-SET (OPENNS_PACKAGE_VERSION_MINOR "${OPENNS_VERS_RELEASE}")
-SET (OPENNS_PACKAGE_STRING "${OPENNS_PACKAGE_NAME} ${OPENNS_PACKAGE_VERSION}")
-SET (OPENNS_PACKAGE_TARNAME "bag")
-SET (OPENNS_PACKAGE_URL "http://www.opennavsurf.org")
-SET (VERSION 1.6.3 OPENNS_PACKAGE_VERSION)
+# SWIG: use SWIG_MODULE_NAME property.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
 
-#-----------------------------------------------------------------------------
-# Option to Build Shared/Static libs, default is shared
-#-----------------------------------------------------------------------------
-OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
-IF (BUILD_SHARED_LIBS)
-  SET (LIB_TYPE SHARED)
-ELSE (BUILD_SHARED_LIBS)
-  SET (LIB_TYPE STATIC)
-ENDIF (BUILD_SHARED_LIBS)
+# Disable addition of warning flags to the CMAKE_<LANG>_FLAGS by default for MSVC.
+if(POLICY CMP0092)
+    cmake_policy(SET CMP0092 NEW)
+endif()
 
-#-----------------------------------------------------------------------------
-# Option to Build against Shared/Static version of HDF5, default is shared
-#-----------------------------------------------------------------------------
-OPTION (BUILD_WITH_SHARED_HDF5 "Build with HDF5 as a shared library" ON)
+project(OPENNS LANGUAGES C CXX VERSION 2.0.0)
 
+# Options
+option(BAG_BUILD_SHARED_LIBS "Build Shared Libraries" ON)
+option(BAG_BUILD_SWIG "Build SWIG interface" OFF)
+option(BAG_BUILD_PYTHON "Build Python bindings" OFF)
```

Comment:
Isn't confusing to have 2 options (BAG_BUILD_SWIG and BAG_BUILD_PYTHON) required to be ON to have built the Python bindings?

I understand the future need for other bindings, but the BAG_BUILD_SWIG should be managed internally based on user selection.

Reply:
Agreed.